### ISSUE TITLE
[sigh] manage doesn't accept -e parameter

### ISSUE THREAD

**Issue by kambala-decapitator** [fastlane/fastlane#18593]
duplicate of #16642

### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

`sigh manage` fails when passing `-e` option. Passing the long variant `--clean_expired` as well as `-e -v` does work though.

##### Command executed

`bundle exec fastlane sigh manage -e`

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>bundler: failed to load command: fastlane (/Users/kambala/.bundler_gems/ruby/2.6.0/bin/fastlane)
OptionParser::MissingArgument: missing argument: -e
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:411:in `parse_global_options'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/fastlane-2.179.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:50:in `run!'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/fastlane-2.179.0/sigh/lib/sigh/commands_generator.rb:139:in `run'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/fastlane-2.179.0/sigh/lib/sigh/commands_generator.rb:16:in `start'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/fastlane-2.179.0/fastlane/lib/fastlane/cli_tools_distributor.rb:114:in `take_off'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/fastlane-2.179.0/bin/fastlane:23:in `<top (required)>'
  /Users/kambala/.bundler_gems/ruby/2.6.0/bin/fastlane:23:in `load'
  /Users/kambala/.bundler_gems/ruby/2.6.0/bin/fastlane:23:in `<top (required)>'</pre>
</details>

### Environment

<details>
  <pre><details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.7                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | true                                                           |
| Git                         | git version 2.24.3 (Apple Git-128)                             |
| Installation Source         | ~/.bundler_gems/ruby/2.6.0/bin/fastlane                        |
| Host                        | Mac OS X 10.15.7 (19H524)                                      |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Library/Developer/CommandLineTools/                           |
| Xcode Version               |                                                                |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.179.0 | 🚫 Update available |

### Loaded fastlane plugins:

| Plugin                    | Version | Update-Status |
| ------------------------- | ------- | ------------- |
| fastlane-plugin-appcenter | 1.11.0  | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                           | Version      |
| ----------------------------- | ------------ |
| did_you_mean                  | 1.3.0        |
| bundler                       | 2.1.4        |
| rake                          | 13.0.3       |
| CFPropertyList                | 3.0.3        |
| concurrent-ruby               | 1.1.8        |
| i18n                          | 1.8.9        |
| minitest                      | 5.14.4       |
| thread_safe                   | 0.3.6        |
| tzinfo                        | 1.2.9        |
| activesupport                 | 5.2.5        |
| public_suffix                 | 4.0.6        |
| addressable                   | 2.7.0        |
| httpclient                    | 2.8.3        |
| json                          | 2.5.1        |
| algoliasearch                 | 1.27.5       |
| artifactory                   | 3.0.15       |
| atomos                        | 0.1.3        |
| aws-eventstream               | 1.1.1        |
| aws-partitions                | 1.436.0      |
| aws-sigv4                     | 1.2.3        |
| jmespath                      | 1.4.0        |
| aws-sdk-core                  | 3.113.0      |
| aws-sdk-kms                   | 1.43.0       |
| aws-sdk-s3                    | 1.93.0       |
| babosa                        | 1.0.4        |
| claide                        | 1.0.3        |
| fuzzy_match                   | 2.0.4        |
| nap                           | 1.1.0        |
| netrc                         | 0.11.0       |
| ffi                           | 1.15.0       |
| ethon                         | 0.12.0       |
| typhoeus                      | 1.4.0        |
| cocoapods-core                | 1.10.1       |
| cocoapods-deintegrate         | 1.0.4        |
| cocoapods-downloader          | 1.4.0        |
| cocoapods-plugins             | 1.0.0        |
| cocoapods-search              | 1.0.0        |
| cocoapods-trunk               | 1.5.0        |
| cocoapods-try                 | 1.2.0        |
| colored2                      | 3.1.2        |
| escape                        | 0.0.4        |
| fourflusher                   | 2.3.1        |
| gh_inspector                  | 1.1.3        |
| molinillo                     | 0.6.6        |
| ruby-macho                    | 1.4.0        |
| nanaimo                       | 0.3.0        |
| xcodeproj                     | 1.19.0       |
| cocoapods                     | 1.10.1       |
| colored                       | 1.2          |
| highline                      | 1.7.10       |
| commander-fastlane            | 4.4.6        |
| declarative                   | 0.0.20       |
| declarative-option            | 0.1.0        |
| digest-crc                    | 0.6.3        |
| unf_ext                       | 0.0.7.7      |
| unf                           | 0.1.4        |
| domain_name                   | 0.5.20190701 |
| dotenv                        | 2.7.6        |
| emoji_regex                   | 3.2.2        |
| excon                         | 0.79.0       |
| faraday-net_http              | 1.0.1        |
| multipart-post                | 2.0.0        |
| ruby2_keywords                | 0.0.4        |
| faraday                       | 1.3.0        |
| http-cookie                   | 1.0.3        |
| faraday-cookie_jar            | 0.0.7        |
| faraday_middleware            | 1.0.0        |
| fastimage                     | 2.2.3        |
| jwt                           | 2.2.2        |
| memoist                       | 0.16.2       |
| multi_json                    | 1.15.0       |
| os                            | 1.1.1        |
| signet                        | 0.15.0       |
| googleauth                    | 0.16.0       |
| mini_mime                     | 1.0.3        |
| uber                          | 0.1.0        |
| representable                 | 3.0.4        |
| retriable                     | 3.1.2        |
| google-api-client             | 0.38.0       |
| rexml                         | 3.2.4        |
| webrick                       | 1.7.0        |
| google-apis-core              | 0.3.0        |
| google-apis-iamcredentials_v1 | 0.2.0        |
| google-apis-storage_v1        | 0.3.0        |
| google-cloud-env              | 1.5.0        |
| google-cloud-errors           | 1.1.0        |
| google-cloud-core             | 1.6.0        |
| google-cloud-storage          | 1.31.0       |
| mini_magick                   | 4.11.0       |
| naturally                     | 2.2.1        |
| plist                         | 3.6.0        |
| rubyzip                       | 2.3.0        |
| security                      | 0.1.3        |
| simctl                        | 1.6.8        |
| slack-notifier                | 2.3.2        |
| terminal-notifier             | 2.0.0        |
| unicode-display_width         | 1.7.0        |
| terminal-table                | 1.8.0        |
| tty-screen                    | 0.8.1        |
| tty-cursor                    | 0.7.1        |
| tty-spinner                   | 0.9.3        |
| word_wrap                     | 1.0.0        |
| rouge                         | 2.0.7        |
| xcpretty                      | 0.3.0        |
| xcpretty-travis-formatter     | 1.0.1        |
| fastlane-plugin-appcenter     | 1.11.0       |
| ruby-lokalise-api             | 4.1.0        |
| xcode-install                 | 2.6.8        |
</details>

*generated on:* **2021-04-23**
</details></pre>
</details>

**Comment by kambala-decapitator** [comment]
added `fastlane env`

please don't tell me to update to the very latest version because I know that it's not solved there.

**Comment by minuscorp** [comment]
In getting the feeling that the `-e` argument if getting overridden by the global option `--env` which makes sense to have precedence. There are other sigh calls that accept a `-e` parameter, you could try one of them to check if it's something isolated with `manage` or not.

**Comment by pedro380085** [comment]
I'm having a similar issue.

**Comment by csc-EricWu** [comment]
I have the same problem

**Comment by minuscorp** [comment]
This is only happening with `manage`? Could someone test what I wrote about `resign` for example and test if it has some relationship with the `-e` parameter?

**Comment by kambala-decapitator** [comment]
@minuscorp at least error isn't triggered, probably because `resign`'s `-e` requires a value:
```
❯ bundle exec fastlane sigh resign -e Gemfile
[✔] 🚀 
[12:37:15]: Path to ipa file:
```

if value isn't provided, it fails (which is correct):
```
❯ bundle exec fastlane sigh resign -e                  
[✔] 🚀 
bundler: failed to load command: fastlane (/Users/kambala/.bundler_gems/ruby/2.6.0/bin/fastlane)
OptionParser::MissingArgument: missing argument: -e
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:411:in `parse_global_options'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/fastlane-2.166.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:50:in `run!'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/fastlane-2.166.0/sigh/lib/sigh/commands_generator.rb:139:in `run'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/fastlane-2.166.0/sigh/lib/sigh/commands_generator.rb:16:in `start'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/fastlane-2.166.0/fastlane/lib/fastlane/cli_tools_distributor.rb:111:in `take_off'
  /Users/kambala/.bundler_gems/ruby/2.6.0/gems/fastlane-2.166.0/bin/fastlane:23:in `<top (required)>'
  /Users/kambala/.bundler_gems/ruby/2.6.0/bin/fastlane:23:in `load'
  /Users/kambala/.bundler_gems/ruby/2.6.0/bin/fastlane:23:in `<top (required)>'
```

**Comment by csc-EricWu** [comment]
fastlane sigh manage -e --verbose 
It's good that I try

**Comment by minuscorp** [comment]
So it clearly is about arguments they expect a type afterwards in its short form what makes it fail... Interesting, I'll try to dig more into this.

**Comment by ainame** [comment]
There's no guarantee but I'm working on updating `commander` gem for Ruby 3.0 support so that might or might not help this problem. I'll let you know if it works with the latest version.

**Comment by kambala-decapitator** [comment]
still the same error in 2.184.0

**Comment by kambala-decapitator** [comment]
still the same error in 2.189.0

**Comment by kambala-decapitator** [comment]
will be fixed by #18760
